R-package/R/lgb.Booster.R

```diff
@@ -713,6 +713,23 @@ Booster <- R6::R6Class(
 #' @param object Object of class \code{lgb.Booster}
 #' @param newdata a \code{matrix} object, a \code{dgCMatrix} object or
 #'                a character representing a path to a text file (CSV, TSV, or LibSVM)
+#' @param type Type of prediction to output. Allowed types are:\itemize{
```

Comment:
Can you please add a note to this documentation that when choosing `"link"` and `"response"`, if you're using a custom objective function they'll be ignored and `"raw"` predictions will be returned?

On the Python side, `lightgbm` raises a warning in such situations.

https://github.com/microsoft/LightGBM/blob/f715645725bcfd7750b9f0a535ef9fd67454eb16/python-package/lightgbm/sklearn.py#L1064-L1067

The R package should probably do that to, but that could be deferred to a later PR.